jetbrains-core/src/software/aws/toolkits/jetbrains/utils/execution/steps/CliBasedStep.kt

```diff
@@ -84,13 +77,15 @@ abstract class CliBasedStep : Step() {
         }
     }
 
-    protected abstract fun handleSuccessResult(output: String, messageEmitter: MessageEmitter, context: Context)
+    protected open fun handleSuccessResult(output: String, messageEmitter: MessageEmitter, context: Context) {}
 
     /**
      * Processes the command's stdout and throws an exception after the CLI exits with failure.
      * @return null if the failure should be ignored. You're probably doing something wrong if you want this.
      */
-    protected abstract fun handleErrorResult(output: String, messageEmitter: MessageEmitter): Nothing?
+    protected open fun handleErrorResult(exitCode: Int, output: String, messageEmitter: MessageEmitter): Nothing? {
+        throw IllegalStateException("Command did not exist successfully, exit code: $exitCode \n")
```

Comment:
why are we putting a `\n` directly into the message?

Reply:
MessageEmitter doesnt add new lines, but this should be localized and have %n in it